File: examples/ospf-kotlin-example/src/main/fuookami/ospf/kotlin/example/framework_demo/demo2/domain/aircraft/model/Fuselage.kt
package fuookami.ospf.kotlin.example.framework_demo.demo2.domain.aircraft.model

import fuookami.ospf.kotlin.utils.math.*
import fuookami.ospf.kotlin.utils.physics.quantity.*

data class Liferaft(
    val weight: Quantity<Flt64>,
    val index: Quantity<Flt64>
)

data class Fuselage(
    val liferaft: Liferaft?,
    val dow: Quantity<Flt64>,
    val doi: Quantity<Flt64>,
    val balancedArm: Quantity<Flt64>,
)

File: ospf-rust-example/src/framework/demo2/domain/aircraft/model/fuselage.rs
use ospf_rust_quantities::quantity::Quantity;
use ospf_rust_quantities::unit::Unit;

/// 救生筏 / Liferaft (对齐 Kotlin Liferaft)
#[derive(Debug, Clone)]
pub struct Liferaft {
    pub weight: Quantity<f64, Unit>,
    pub index: f64,
}

/// 机身 / Fuselage (对齐 Kotlin Fuselage)
#[derive(Debug, Clone)]
pub struct Fuselage {
    pub liferaft: Option<Liferaft>,
    pub dow: Quantity<f64, Unit>,       // Dry Operating Weight
    pub doi: f64,                       // Dry Operating Index
    pub balanced_arm: Quantity<f64, Unit>,
}
